src/dbus_api/api/mod.rs

```diff
@@ -0,0 +1,135 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+use std::{cell::RefCell, rc::Rc, vec::Vec};
+
+use dbus::{
+    self,
+    tree::{Factory, MTFn, Tree},
+    BusType, Connection, ConnectionItem, NameFlag,
+};
+use libc;
+
+use crate::{
+    dbus_api::{
+        api::manager_2_0::api::{
+            configure_simulator_method, create_pool_method, destroy_pool_method, version_property,
+        },
+        blockdev::create_dbus_blockdev,
+        consts,
+        filesystem::create_dbus_filesystem,
+        pool::create_dbus_pool,
+        types::{DbusContext, DeferredAction, TData},
+    },
+    engine::{Engine, Pool, PoolUuid},
+};
+
+mod manager_2_0;
+
+fn get_base_tree<'a>(dbus_context: DbusContext) -> (Tree<MTFn<TData>, TData>, dbus::Path<'a>) {
+    let f = Factory::new_fn();
+
+    let base_tree = f.tree(dbus_context);
+
+    let obj_path = f
+        .object_path(consts::STRATIS_BASE_PATH, None)
+        .introspectable()
+        .object_manager()
+        .add(
+            f.interface(consts::MANAGER_INTERFACE_NAME, ())
+                .add_m(create_pool_method(&f))
+                .add_m(destroy_pool_method(&f))
+                .add_m(configure_simulator_method(&f))
+                .add_p(version_property(&f)),
+        );
+
+    let path = obj_path.get_name().to_owned();
+    (base_tree.add(obj_path), path)
+}
+
+/// Returned data from when you connect a stratis engine to dbus.
```

Comment:
Might as well move all DbusConnectionData code into a separate connection module while you're here. I don't think it every really belonged in the api module anyway.